client/src/context.test.js

```diff
@@ -0,0 +1,58 @@
+import { beforeEach, describe, expect, it, jest } from "@jest/globals";
+import { renderHook, act } from "@testing-library/react-hooks";
+import { FetchContextProvider, useFetchContext } from "./context";
+
+describe("useFetchContext", () => {
```

Comment:
Should tests be added for getText and getJSON? Should these tests cover the error cases that I expressed concerns about above?